lib/screens/detail/nps_answer.dart

```diff
@@ -0,0 +1,131 @@
+import 'package:flutter/material.dart';
+import 'package:survey_flutter/model/survey_answer_model.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
+class NpsAnswer extends StatefulWidget {
+  final ValueChanged<String> onChoiceClick;
+  final List<SurveyAnswerModel> answers;
+
+  const NpsAnswer({
+    required this.onChoiceClick,
+    required this.answers,
+    super.key,
+  });
+
+  @override
+  State<StatefulWidget> createState() => NpsAnswerState();
+}
+
+class NpsAnswerState extends State<NpsAnswer> {
+  TextTheme get _textTheme => Theme.of(context).textTheme;
+
+  AppLocalizations get _localizations => AppLocalizations.of(context)!;
+  int? _selectedIndex;
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
```

Comment:
I think we should make a flag here and use it for below code? 

```suggestion
    final isNotAtAllLikely = (_selectedIndex != null &&
                      (_selectedIndex ?? 0) <
                          widget.answers.length / 2.round())
```

Reply:
@ducbm051291 They are not duplicated. They both have to check whether the `_selectedIndex` is `null` or not at first and the rest is their condition. IMO I don't think making a flag here could a lot of difference here. 🙏